R/UnitConversions.R

```diff
@@ -24,17 +24,27 @@
 #'
 #' @param transform Boolean argument with two possible values, “TRUE” and “FALSE”.
 #' Default is transform = TRUE.
+#' 
+#' @param detlimit Boolean arguement with two possible values, "TRUE" and "FALSE".
+#' Default is detlimit = TRUE.
 #'
 #' @return When transform=TRUE, result values and units are converted to WQX
 #'   target units. This function changes the values within the
 #'   "TADA.ResultMeasure.MeasureUnitCode" to the WQX target units and converts
 #'   respective values within the "TADA.ResultMeasureValue" field.
+#'   When detlimit = TRUE, detection limit values and units are converted to WQX
+#'   target units. This function changes the 
+#'   "TADA.DetectionQuantitationLimitMeasure.MeasureUnitCode" to the WQX target
+#'   units and converts respective values within the 
+#'   "TADA.DetectionQuantitationLimitMeasure.MeasureValue" field. 
 #'
 #' When transform = FALSE, result values and units are NOT converted to WQX target units,
 #' but columns are appended to indicate what the target units and conversion factors are,
 #' and if the data can be converted. In addition to "TADA.WQXResultUnitConversion"
 #' and "TADA.WQXDetectionLimitUnitConversion",  transform=FALSE will add the
```

Comment:
See example below where "TADA.WQXResultUnitConversion" says "No Result Value", but the result value in the tada result field is populated (copied from del limit). 

If we decide to keep "TADA.WQXResultUnitConversion" then we can address this in the ID censored data function. 

``` r
 df = TADA_DataRetrieval(startDate = "2022-12-19",
                                 endDate = "2022-12-20")

df1 = TADA_IDCensoredData(df)
  
df2 = subset(df1, subset = df1$ResultMeasureValue == "BPQL" 
                 | df1$ResultMeasureValue == "BDL"
                 | df1$ResultMeasureValue == "ND")
  
unique(df2$ResultMeasureValue)
  
# subset df: TADA.DetectionQuantitationLimitMeasure.MeasureValue = NA or None
df3 = subset(df2, subset = (!is.na(df2$TADA.DetectionQuantitationLimitMeasure.MeasureValue)))
  
df3_subset = dplyr::select(df3, 
                            ActivityTypeCode, 
                            TADA.ActivityType.Flag,
                            ResultDetectionConditionText,
                            CharacteristicName,
                            TADA.CharacteristicName,
                            ResultMeasureValue,
                            TADA.ResultMeasureValue,
                            ResultMeasure.MeasureUnitCode,
                            TADA.WQXResultUnitConversion,
                            # TADA.WQXDetectionLimitUnitConversion, 
                            TADA.ResultMeasureValueDataTypes.Flag,
                            TADA.DetectionQuantitationLimitMeasure.MeasureValueDataTypes.Flag,
                            TADA.ResultMeasure.MeasureUnitCode,
                            DetectionQuantitationLimitTypeName,
                            DetectionQuantitationLimitMeasure.MeasureValue,
                            DetectionQuantitationLimitMeasure.MeasureUnitCode,
                            TADA.DetectionQuantitationLimitMeasure.MeasureValue,
                            TADA.DetectionQuantitationLimitMeasure.MeasureUnitCode,
                            ProviderName)
```

Reply:
Good catch! I had initially planned to add a TADA.WQXDetectionLimitUnitConversion column, but then ultimately didn't as I didn't think it provided much (any?)additional useful information.

I would lean towards TADA.WQXResultUnitConversion not always being necessary.

Reply:
I agree. Let's adjust so that "TADA.WQXResultUnitConversion" is only added if transform = FALSE like the other three. This would mean four columns total are added ONLY if transform = FALSE.

Reply:
Do you want me to make that adjustment? And remove erroneous reference to TADA.WQXDetectionLimitUnitConversion?